libbeat/common/schema/mapstriface/mapstriface.go

```diff
@@ -91,30 +90,31 @@ type ConvMap struct {
 }
 
 // Map drills down in the data dictionary by using the key
-func (convMap ConvMap) Map(key string, event mapstr.M, data map[string]interface{}) multierror.Errors {
+func (convMap ConvMap) Map(key string, event mapstr.M, data map[string]interface{}) []error {
 	d, err := mapstr.M(data).GetValue(convMap.Key)
 	if err != nil {
 		err := schema.NewKeyNotFoundError(convMap.Key)
 		err.Optional = convMap.Optional
 		err.Required = convMap.Required
-		return multierror.Errors{err}
+		return []error{err}
 	}
 	switch subData := d.(type) {
 	case map[string]interface{}, mapstr.M:
 		subEvent := mapstr.M{}
-		_, errors := convMap.Schema.ApplyTo(subEvent, subData.(map[string]interface{}))
-		for _, err := range errors {
-			if err, ok := err.(schema.KeyError); ok {
-				err.SetKey(convMap.Key + "." + err.Key())
+		_, errs := convMap.Schema.ApplyTo(subEvent, subData.(map[string]interface{}))
+		for _, err := range errs {
+			var keyErr schema.KeyError
```

Comment:
This is surprising. It would be made less so if there were a comment explaining that while `schema.KeyError` doesn't satisfy `error`, its implementations do.